cfdm/data/data.py

```diff
@@ -1901,6 +2046,150 @@ def minimum(self, axes=None):
 
         return out
 
+    def original_filenames(self, define=None, update=None, clear=False):
+        """Return the names of files that contain the original data.
+
+        The original files are those that contain some or all of the
+        data when it was first instantiated.
+
+        The original files are necessary (but perhaps not sufficient)
+        to recreate the data, should the need arise.
+
+        If the data was produced by combining other objects that also
+        store their original file names, then the returned files will
+        be the collection of original files from all contributing
+        sources.
+
+        If there are any ancillary data information, such as a count
+        variable required for compressed data, then their original
+        files are also included in the returned values.
+
+        .. versionadded:: (cfdm) 1.10.0.1
+
+        .. seealso:: `get_filenames`
+
+        :Parameters:
+
+            define: (sequence of) `str`, optional
+                Set these original file names, removing any already
+                stored.
```

Comment:
Something to indicate explicitly that only one of the three keyword parameters can/should be set, i.e. not any two or all three, might be useful, even if it should be clear enough from the descriptions.

Reply:
I'm happy leaving this to be trapped in the code:
```python
                raise ValueError(
                    "Can't set the 'define' and 'update' parameters "
                    "at the same time"
```
```python
                raise ValueError(
                    "Can't set the 'clear' parameter with either of the "
                    "'define' and 'update' parameters"
                )
```
Is that OK with you? Happy to doc it if you prefer.

Reply:
I'm inclined to say I think it should also be documented, even if that is to very concisely indicate as such, but it's your call!

Reply:
OK - will do. I'll hold off implementing it until we're agreed on where we stand on https://github.com/NCAS-CMS/cf-python/issues/448#issuecomment-1266560596